src/Core/Grid/Query/OrderReturnStatusesQueryBuilder.php

```diff
@@ -0,0 +1,185 @@
+        $contextLangId,
+        array $contextShopIds
+    ) {
+        parent::__construct($connection, $dbPrefix);
+
+        $this->contextLangId = $contextLangId;
+        $this->contextShopIds = $contextShopIds;
+        $this->criteriaApplicator = $criteriaApplicator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $searchQueryBuilder = $this->getOrderReturnStatusesQueryBuilder($searchCriteria)
+            ->select('ors.id_order_return_state, orsl.name, ors.color');
+
+        $this->applySorting($searchQueryBuilder, $searchCriteria);
+
+        $this->criteriaApplicator->applyPagination(
+            $searchCriteria,
+            $searchQueryBuilder
+        );
+
+        return $searchQueryBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $countQueryBuilder = $this->getOrderReturnStatusesQueryBuilder($searchCriteria)
+            ->select('COUNT(*)');
+
+        return $countQueryBuilder;
+    }
+
+    /**
+     * @param SearchCriteriaInterface $searchCriteria
+     *
+     * @return QueryBuilder
+     */
+    private function getOrderReturnStatusesQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $queryBuilder = $this->connection->createQueryBuilder()
+            ->from($this->dbPrefix . 'order_return_state', 'ors')
+            ->leftJoin(
+                'ors',
+                $this->dbPrefix . 'order_return_state_lang',
+                'orsl',
+                'ors.id_order_return_state = orsl.id_order_return_state AND orsl.id_lang = :context_lang_id'
+            )
+            ->setParameter('context_lang_id', $this->contextLangId);
+
+        $this->applyFilters($searchCriteria->getFilters(), $queryBuilder);
+
+        return $queryBuilder;
+    }
+
+    /**
+     * Apply filters to order_statuses query builder.
+     *
+     * @param array $filters
+     * @param QueryBuilder $qb
+     */
+    private function applyFilters(array $filters, QueryBuilder $qb)
+    {
+        $allowedFilters = [
+            'id_order_return_state',
+            'name',
+        ];
+
+        foreach ($filters as $filterName => $filterValue) {
+            if (!in_array($filterName, $allowedFilters)) {
+                continue;
+            }
+
+            if ('id_order_return_state' === $filterName) {
+                $qb->andWhere('ors.`' . $filterName . '` = :' . $filterName);
+                $qb->setParameter($filterName, $filterValue);
+
+                continue;
+            }
+
+            if ('name' === $filterName) {
+                $qb->andWhere('orsl.`' . $filterName . '` LIKE :' . $filterName);
+                $qb->setParameter($filterName, '%' . $filterValue . '%');
+
+                continue;
+            }
+
+            $qb->andWhere('`' . $filterName . '` LIKE :' . $filterName);
```

Comment:
This is not used. Both lines can be removed

Reply:
:+1:  done